lib/std/Thread.zig

```diff
@@ -831,15 +834,42 @@ const WasiThreadImpl = struct {
         }
     }
 
-    fn spawn(config: std.Thread.SpawnConfig, comptime f: anytype, args: anytype) !WasiThreadImpl {
-        if (config.allocator == null) return error.OutOfMemory; // an allocator is required to spawn a WASI-thread
+    fn spawn(config: std.Thread.SpawnConfig, comptime f: anytype, args: anytype) SpawnError!WasiThreadImpl {
+        if (config.allocator == null) {
+            return error.AllocatorRequired; // an allocator is required to spawn a WASI thread
```

Comment:
Shouldn't this be a panic since it is never valid to not pass an allocator?

Reply:
Returning an error was the original behavior, but yeah, we should just panic.

Changed. Thanks!